frontend/src/types/index.ts

```diff
@@ -0,0 +1,15 @@
+export interface AppState {
+  auth: Auth
+}
+
+export interface Auth {
+  is_refreshing: boolean
+  auth_token: string,
+  refresh_token: string
+  last_refresh?: Date
+}
+
+export interface ReduxAction<T,P> {
```

Comment:
I'd argue for just calling this bad boy `Action`-- e.g. we use `AppState`, not `ReduxAppState`.

Reply:
👍